pennylane/qchem/hamiltonian.py

```diff
@@ -170,21 +175,30 @@ def _fermionic_hamiltonian(*args):
             *args (array[array[float]]): initial values of the differentiable parameters
 
         Returns:
-            tuple(array[float], list[list[int]]): the Hamiltonian coefficients and operators
+            Union[FermiSentence, tuple(array[float], list[list[int]])]: fermionic Hamiltonian
         """
+
         core_constant, one, two = electron_integrals(mol, core, active)(*args)
 
-        return fermionic_observable(core_constant, one, two, cutoff)
+        if not fs:
+            warnings.warn(
+                "This function will return a fermionic operator by default in the next release. For"
+                " details, see the Fermionic Operators tutorial:"
+                " https://pennylane.ai/qml/demos/tutorial_fermionic_operators"
+            )
+        return fermionic_observable(core_constant, one, two, cutoff, fs)
 
     return _fermionic_hamiltonian
 
 
-def diff_hamiltonian(mol, cutoff=1.0e-12, core=None, active=None):
+def diff_hamiltonian(mol, cutoff=1.0e-12, core=None, active=None, fs=True):
```

Comment:
> What is the reasoning for defaulting to True here, and False above? I would have expected that we leave the default behaviour as it has been in the past (but with a warning) for one release cycle before we change the default output type for a function.

This particular function calls `fermionc_hamiltonian` to construct a fermionic Hamiltonian, and then converts it to a PL Hamiltonian. Because creating the fermionic Hamiltonia is an internal task, we can already build a Fermi sentence and benefit from creating and mapping it to a PL Hamiltonian without the users noticing any change. So, I think it is safe that the default is `True` here, but open to reconsider. 

> Assuming the default should be True, the example in the docstring needs the return updated as well

In this case no, because the output is a PL Hamiltonian, regardless of how we create the fermionic one internally in this function.

Reply:
The numbers are the same, but the output format has changed a little, to be a list of individual tensors of length 1, instead of one longer tensor. Not sure where this comes from, just something I noticed when running the examples:

```
[tensor(0.29817878+0.j, requires_grad=True),
 tensor(0.20813366+0.j, requires_grad=True),
 tensor(-0.34724872+0.j, requires_grad=True),
 tensor(0.13290292+0.j, requires_grad=True),
 tensor(0.20813366+0.j, requires_grad=True),
 tensor(0.17860977+0.j, requires_grad=True),
 tensor(0.04256036+0.j, requires_grad=True),
 tensor(-0.04256036+0.j, requires_grad=True),
 tensor(-0.04256036+0.j, requires_grad=True),
 tensor(0.04256036+0.j, requires_grad=True),
 tensor(-0.34724872+0.j, requires_grad=True),
 tensor(0.17546328+0.j, requires_grad=True),
 tensor(0.13290292+0.j, requires_grad=True),
 tensor(0.17546328+0.j, requires_grad=True),
 tensor(0.18470917+0.j, requires_grad=True)]
```

Reply:
Sounds like this default behaviour makes sense :)